apps/web/components/device-auth/device-code-form.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from "@/components/ui/form";
+import { Input } from "@/components/ui/input";
+import { api } from "@/trpc/react";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Check, Loader2 } from "lucide-react";
+import Link from "next/link";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+// Device code format: 8 alphanumeric characters with optional dash (e.g., "WDJB-MJHT" or "WDJBMJHT")
+const deviceCodeSchema = z.object({
+  userCode: z
+    .string()
+    .min(1, "Please enter a device code")
+    .transform((val) => val.replace(/-/g, "").trim())
+    .pipe(
+      z
+        .string()
+        .length(8, "Device code must be 8 characters")
+        .regex(/^[A-Z0-9]+$/i, "Device code must be alphanumeric"),
```

Comment:
if you're going to use a regex, might as well use something that validates the exact format, e.g. `/[A-Z0-9]{4}-[A-Z0-9]{4}/i`